Add Python binding lib.rs and fix Ruby publishing working directory

## Description

The release workflow has two remaining issues:

### 1. PyPI - Missing lib.rs for Python bindings

**Error:**
```
error: failed to parse manifest at /home/runner/work/jcl/jcl/bindings/python/Cargo.toml
Caused by: can't find library jcl, rename file to src/lib.rs or specify lib.path
```

**Root Cause:** The `bindings/python/Cargo.toml` declares a cdylib library but there's no `bindings/python/src/lib.rs`. The Python binding code exists at `src/bindings/python.rs` in the main crate.

**Solution:** Create `bindings/python/src/lib.rs` that re-exports the Python bindings from the parent crate.

### 2. RubyGems - Wrong working directory

**Error:**
```
Run bundle exec rake release
Could not locate Gemfile or .bundle/ directory
Error: Process completed with exit code 10.
```

**Root Cause:** The `rubygems/release-gem@v1` action needs to run from `bindings/ruby/` directory where the gemspec and gem file are located.

**Solution:** Add `working-directory: bindings/ruby` to the RubyGems publishing step.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -242,4 +242,6 @@ jobs:
 
       - name: Publish to RubyGems (with trusted publishing)
         uses: rubygems/release-gem@v1
+        with:
+          working-directory: bindings/ruby
 

--- a/bindings/python/src/lib.rs
+++ b/bindings/python/src/lib.rs
@@ -1,0 +1,3 @@
+// Re-export Python bindings from the parent jcl crate
+// The actual Python binding implementation is in src/bindings/python.rs
+pub use jcl::*;

